Automattic/jetpack
# Social: Remove share limits UI and data logic

Before we can migrate shares data initial state, we need to remove the share limits UI  and the corresponding data store logic, which uses initial state, otherwise we will need to migrate that unused initial state as well as the corresponding logic. Thus we want to remove that first.

## Proposed changes:

* <s>Remove unused `show_advanced_plan_upgrade_nudge` and the related components because it was already disabled in  D151913-code</s> Extracted to #38926
* Remove share limits UI and data store logic

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Goto Social admin page
* Confirm that the page loads correctly and the shares data is displayed as before
* Create a new post and open Jetpack/Social sidebar
* Wait for connection test results to finish
* Confirm that the connections remain enabled after test results, because we previously it depended upon `numberOfSharesRemaining`
